src/pattern-library/components/patterns/navigation/TabPage.js

```diff
@@ -0,0 +1,390 @@
+import classnames from 'classnames';
+import { useState } from 'preact/hooks';
+
+import {
+  Card,
+  EmailIcon,
+  ProfileIcon,
+  SettingsIcon,
+  Tab,
+  TabList,
+} from '../../../../next';
+import Library from '../../Library';
+import Next from '../../LibraryNext';
+
+export default function TabPage() {
+  const [prefPanel, setPrefPanel] = useState('notifications');
+  const [sidebarPanel, setSidebarPanel] = useState('annotations');
+  const [sidebarPanel2, setSidebarPanel2] = useState('annotations');
+  const [sidebarPanel3, setSidebarPanel3] = useState('annotations');
+  const [verticalPanel, setVerticalPanel] = useState('notifications');
+  return (
+    <Library.Page
+      title="Tabs"
+      intro={
+        <p>
+          <code>Tab</code> and <code>TabList</code> are presentational
+          components for rendering accessible tabs.
+        </p>
+      }
+    >
+      <Library.Section
+        title="Tab"
+        intro={
+          <p>
+            <code>Tab</code> generates a button with appropriate ARIA
+            attributes.
+          </p>
+        }
+      >
+        <Library.Pattern title="Status">
+          <p>
+            <code>Tab</code> is a new component.
+          </p>
+        </Library.Pattern>
+
+        <Library.Pattern title="Usage">
+          <Next.Usage componentName="Tab" />
+          <Library.Example>
+            <Library.Demo title="Basic (non-interactive) example" withSource>
+              <div role="tablist" className="gap-x-6 flex">
+                <Tab>
+                  Annotations
+                  <span className="relative bottom-[3px] left-[2px] text-[10px]">
+                    {52}
+                  </span>
+                </Tab>
+                <Tab selected>
+                  Page Notes
+                  <span className="relative bottom-[3px] left-[2px] text-[10px]">
+                    {4}
+                  </span>
+                </Tab>
+                <Tab>
+                  Orphans
+                  <span className="relative bottom-[3px] left-[2px] text-[10px]">
+                    {2}
+                  </span>
+                </Tab>
+              </div>
+            </Library.Demo>
+          </Library.Example>
+
+          <ul>
+            <li>
+              <code>Tab</code>s <em>must</em> be direct children of an element
+              with <code>role={'"tablist"'}</code> (or use the{' '}
+              <code>TabList</code> component).
+            </li>
+            <li>
+              You <em>should</em> provide an{' '}
+              <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls">
+                <code>aria-controls</code>
+              </a>{' '}
+              attribute to each <code>Tab</code>. This is not always feasible in
+              our applications.
```

Comment:
My initial reaction to "This is not always feasible in our applications" was "why?". Are we doing something unusual or is there an implementation problem? What are the problems caused by not using this?

Reply:
Great question! The WAI-ARIA authoring guidelines for a full tabby-widget-y interface assumes  a structure like so:

```
<tablist>
  <tab controls="panel1">One</tab>
  <tab controls="panel2" aria-selected>Two</tab>
  <tab controls="panel3">Three</tab>
</tablist>
<tabpanel id="panel1" style="visibility:hidden">Content for panel One</tabpanel>
<tabpanel id="panel2">Content for panel Two</tabpanel>
<tabpanel id="panel3" style="visibility:hidden">Content for panel Three</tabpanel>
```

To wit:

* A `tabpanel` for each `tab`
* Non-active `tabpanel`s are visually hidden

However, in the sidebar, we have:

```
<tablist>
  <tab>Annotations</tab>
  <tab aria-selected>Page Notes</tab>
  <tab>Orphans</tab>
</tablist>
<div>Page-note annotation threads</div>
```

To wit: All three tabs control the same "panel"; we don't have a discrete element "per panel". I don't think it would appropriate to assign the same `aria-controls` value to each tab (i.e. the `id` of the `div` containing threads, which, for clarity, does not currently have an `id` attribute). But I could be wrong.

I think this sort of situation is why WAI-ARIA guidelines indicate that there _should_ be an `aria-controls` on `role="tab"` elements, but does not indicate that there _must_.

Reply:
> To wit: All three tabs control the same "panel"; we don't have a discrete element "per panel".

Ah, OK. That should be an implementation detail that we are re-using the DOM elements though. From a user's perspective (screen reader or not) I think it would ideally "look" the same as if we were using separate panels. I wonder if there is a blessed way of doing this kind of virtualization. How does `aria-controls` actually get used?